src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp

```diff
@@ -1191,4 +1193,81 @@ namespace TerminalAppLocalTests
             VERIFY_IS_TRUE(caughtExpectedException);
         }
     }
+
+    void SettingsTests::TestLayerGlobalsOnRoot()
+    {
+        // Test for microsoft/terminal#2906. We added the ability for the root
+        // to be used as the globals object in #2515. However, if you have a
+        // globals object, then the settings in the root would get ignored.
+        // This test ensures that settings from a child "globals" element
+        // _layer_ on top of root properties, and they don't cause the root
+        // properties to be totally ignored.
+
+        const std::string settings0String{ R"(
+        {
+            "globals": {
+                "defaultProfile": "{6239a42c-1111-49a3-80bd-e8fdd045185c}",
+                "initialRows": 123
+            }
+        })" };
+        const std::string settings1String{ R"(
+        {
+            "defaultProfile": "{6239a42c-1111-49a3-80bd-e8fdd045185c}",
+            "initialRows": 234
+        })" };
+        const std::string settings2String{ R"(
+        {
+            "defaultProfile": "{6239a42c-2222-49a3-80bd-e8fdd045185c}",
+            "initialRows": 345,
+            "globals": {
+                "defaultProfile": "{6239a42c-1111-49a3-80bd-e8fdd045185c}"
+                // initialRows should not be cleared here
+            }
+        })" };
+        const std::string settings3String{ R"(

```

Comment:
Could you add another test:
```
        const std::string settings4String{ R"(
        {
            "initialRows": 345,
            "globals": {
                "defaultProfile": "{6239a42c-1111-49a3-80bd-e8fdd045185c}"
                // initialRows should not be cleared here
            },
            "defaultProfile": "{6239a42c-2222-49a3-80bd-e8fdd045185c}"
        })" };
```
I think as of now, processing this will make defaultProfile `{6239a42c-1111-49a3-80bd-e8fdd045185c}` instead of `{6239a42c-2222-49a3-80bd-e8fdd045185c}`.

Reply:
Sure, but I think that's the expected behavior with this change unfortunately. Once we've parsed the json, it doesn't retain any of its original ordering any longer. I'm not sure how we'd be able to get the 2222 guid to be the default in that case.

Reply:
> Sure, but I think that's the expected behavior with this change unfortunately. Once we've parsed the json, it doesn't retain any of its original ordering any longer. I'm not sure how we'd be able to get the 2222 guid to be the default in that case.

Think it might be worth creating an issue then and just leaving it for later?

Reply:
Oh, i see: file level ordering. Interesting.

...

we can check getOffsetStart and getOffsetLimit :wink:

Reply:
gosh Dustin why'd you have to suggest that

That's certainly an option, but that's a lot harder. We'd have to do it on a peculiar property-by-property basis. Layering wouldn't just be "do I have this property? great, let's parse it". It's now "for the globals, do I have this property? was its `getOffsetStart` greater than the last time I layered this property?" 

That takes it from a little change to a much, MUCH bigger change